### ISSUE TITLE
Feature Request - Static Batching

### ISSUE THREAD

**Issue by kkafaseb** [cocos2d/cocos2d-x#19389]
Static Batching: This is a feature that is supported in Unity. If i understand correctly, you can mark certain objects as static and although not sharing same properties they are grouped together as one texture internally therefore minimizing gl calls. Any way to support this feature?

**Comment by minggo** [comment]
I think you can use SpriteBatchNode for it.

**Comment by kkafaseb** [comment]
@minggo  oh really? I thought SpriteBatchNode needed same textures to work and was also deprecated. So even with different textures, it will still batch?

**Comment by minggo** [comment]
Different textures can not be batched.

**Comment by kkafaseb** [comment]
exactly. so i think unity static batch can combine internally different textures into 1 bigger. got it?

**Comment by sebas86** [comment]
No, Unity was never merge textures for static batches. It just merge meshes as many as it can to create one vertex buffer, any single material has own batch. If you wan't bigger batches in unity you should merge textures by hand. BTW texture automerging is inefficient and error prone... this approach can exceed optimum texture size also there are other problems.

**Comment by kkafaseb** [comment]
Thanks for calirification sebas86

**Comment by imtrobin** [comment]
Actually, Unity does auto pack, it's just not enabled by default in Editor

https://docs.unity3d.com/Manual/SpritePacker.html

**Comment by kkafaseb** [comment]
Also Unity static batching here: https://docs.unity3d.com/Manual/DrawCallBatching.html 

Basically takes dynamic batching which is already supported in cocos, a step further: 

![screenshot 2019-02-18 at 2 35 39 pm](https://user-images.githubusercontent.com/42576559/52951481-9207d100-338a-11e9-9caa-c0f0ba290e67.png)

**Comment by Nordsoft91** [comment]
Also need this feature.
